src/memray/_memray/alloc.h

```diff
@@ -0,0 +1,10 @@
+#include <malloc.h>
+#include <stdlib.h>
+
+#ifndef __GLIBC__
+void*
+pvalloc(size_t size)
```

Comment:
This would break and require a change to our source if musl ever added `pvalloc`. But fixing that requires some sort of configure-like system, or feature test macros, or something... 😕

Reply:
If you want to handle this differently, I am open to suggestions :)

Reply:
Technically we just need this so the test suite works, but maybe there are other better ways to do this...

Reply:
The test suite probably shouldn't depend on a function that even the glibc docs document as "obsolete", to be fair...

We're also not intercepting `aligned_alloc` right now, which the glibc docs suggest as a replacement for the obsolete `valloc`... I opened #78 for that.

Reply:
Alright, I can't immediately think of any better idea than this, but - doesn't this need an `extern "C"` on it?

Reply:
We *could* just not care about `pvalloc` to be honest....

Reply:
I think it's fine, let's just add the `extern "C"`